src/Microsoft.Health.Fhir.Web/Startup.cs

```diff
@@ -24,9 +25,16 @@ public virtual void ConfigureServices(IServiceCollection services)
         {
             services.AddDevelopmentIdentityProvider(Configuration);
 
-            services.AddFhirServer(Configuration)
-                .AddExportWorker()
-                .AddCosmosDb(Configuration);
+            Core.Registration.IFhirServerBuilder fhirServerBuilder = services.AddFhirServer(Configuration).AddExportWorker();
+
+            if (Configuration["DataStore"].Equals(KnownDataStores.CosmosDb, StringComparison.InvariantCultureIgnoreCase))
+            {
+                fhirServerBuilder.AddCosmosDb(Configuration);
+            }
+            else if (Configuration["DataStore"].Equals(KnownDataStores.SqlServer, StringComparison.InvariantCultureIgnoreCase))
```

Comment:
I wonder if this process could be a little bit more automatic. E.g., some class with some attribute to indicate that data store type and the code automatically discover and register it based on that attribute?